src/index-editor.js

```diff
@@ -79,6 +79,14 @@ export default class CalendarCardEditor extends LitElement {
             >Show Location Icon</paper-checkbox>
           </div>
 
+          <div class='checkbox-options'>
+            <paper-checkbox
+              @checked-changed="${this._valueChanged}" 
+              .checked=${this._config.showCalendarName}
```

Comment:
There are 2 ways currently:

```
entities:
  - calendar.example_calendar
```
in here the name will be example_calendar

```
entities:
  - entity: calendar.example_calendar
    name: Example calendar
```
in here the name will be Example calendar

I honestly don't know how the index-editor works so I don't know how to make your requested changes.

Reply:
yeah my biggest worry is backwards compatibility since 'older' configs will be strings but newer ones will be objects. I used the layout of what you did here to add the functionality of hat you want to v3.11.0

Reply:
Cool! Thanks :)